src/Microsoft.Graph.Core/Serialization/TimeOfDayConverter.cs

```diff
@@ -39,19 +39,18 @@ public override TimeOfDay Read(ref Utf8JsonReader reader, Type typeToConvert, Js
         {
             try
             {
-                var dateTime = (DateTime)JsonSerializer.Deserialize(reader.GetString(), typeof(DateTime));
-
+                var dateTime = DateTime.Parse(reader.GetString());
```

Comment:
```suggestion
                var dateTime = reader.GetDateTime();
```

Reply:
@baywet This will also lead to the throwing of an exception if the datetime also does not conform to the ISO 8601-1:2019 date format.
See [here](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.getdatetime?view=net-5.0#remarks)

Reply:
I wasn't suggesting it wouldn't throw an exception, just that there's already a method to get a datetime and we should use it instead of doing the parsing ourselves. Unless something is blocking with the way that method is implemented.

Reply:
We will need to do the parsing ourselves though since the response payload of a TimeOfDay instance may not comply to the standard. 
An example string from issue #244 is "startTime": "08:00:00.0000000" which doesn't comply.

Reply:
oooohhhh I get it now... sorry about that. And utf8jsonreader doesn't provide any configuration capability for that aspect as far as I can tell. Could you add a comment please?

Reply:
Sure thing will do.

Reply:
resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/245/commits/c0bab2f9f1406f5e75519cebed7a9ec7ab0c2c73